wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/QR Code/Views/QRCodeView.swift

```diff
@@ -0,0 +1,128 @@
+    @State private var isShareImageSheetPresented = false
+    @State private var capturedImage: UIImage?
+    @State private var shouldCaptureImage = false
+
+    // MARK: - View
+
+    var body: some View {
+        GeometryReader { geometry in
+            ScrollView(showsIndicators: false) {
+                shareView
+                    .frame(minHeight: geometry.size.height)
+            }
+        }
+        .background(Color.viewBackground.edgesIgnoringSafeArea(.all))
+        .onChange(of: shouldCaptureImage) { newValue in
+            if newValue {
+                capturedImage = captureImage(from: qrCodeCard)
+                shouldCaptureImage.toggle()
+            }
+        }
+    }
+
+    private var shareView: some View {
+        return VStack {
+            qrCodeCard
+            infoTextView
+            Spacer()
+            shareButtons
+        }
+        .padding()
+    }
+
+    private var qrCodeCard: some View {
+        QRCodeCard(
+            profileLinkQRCode: viewModel.profileLinkQRCode,
+            handle: viewModel.handle,
+            profileLink: viewModel.profileLink
+        )
+    }
+
+    private var infoTextView: some View {
+        Text(L10n.Localizable.Qrcode.shareMessage)
+            .font(.textStyle(.body1))
+            .multilineTextAlignment(.center)
+            .foregroundColor(Color.secondaryText)
+            .fixedSize(horizontal: false, vertical: true)
+    }
+
+    private var shareButtons: some View {
+        VStack {
+            Button(L10n.Localizable.Qrcode.ShareProfileLink.Button.title) {
+                isShareTextSheetPresented = true
+            }
+            .font(.textStyle(.buttonBig))
+            .buttonStyle(SecondaryButtonStyle())
+            .sheet(isPresented: $isShareTextSheetPresented) {
+                ShareSheet(activityItems: [viewModel.profileLink])
+            }
+
+            Button(L10n.Localizable.Qrcode.ShareQrcode.Button.title) {
+                isShareImageSheetPresented = true
+                shouldCaptureImage = true
+            }
+            .font(.textStyle(.buttonBig))
+            .buttonStyle(SecondaryButtonStyle())
+            .sheet(isPresented: $isShareImageSheetPresented) {
+                if let image = capturedImage {
+                    ShareSheet(activityItems: [image])
+                }
+            }
+        }
+    }
+
+}
+
+// MARK: - Capture image
+
+extension QRCodeView {
+
+    private func captureImage<Content: View>(from view: Content) -> UIImage? {
+        let controller = UIHostingController(rootView: view)
+        let targetSize = CGSize(width: 400, height: 400)
+        controller.view.bounds = CGRect(origin: .zero, size: targetSize)
+        controller.view.backgroundColor = .clear
+
+        let renderer = UIGraphicsImageRenderer(size: targetSize)
+        return renderer.image { _ in
+            controller.view.drawHierarchy(in: controller.view.bounds, afterScreenUpdates: true)
+        }
+    }
```

Comment:
regarding the issue with the QR code not showing up when sharing on iOS 16, this code fixes the issue, right ?

Reply:
I've added 2 states:
      @State private var capturedImage: UIImage?
    @State private var shouldCaptureImage = false
    
For some reason, image wasn't captured at the moment we wanted to share in in iOS 16